docs/src/main/asciidoc/_admin-guide/management-tasks/Configuration_file_git_history.adoc

```diff
@@ -0,0 +1,101 @@
+[[Configuration_file_git_history]]
+= Git Configuration file history
+
+To enhance the initial <<Configuration_file_history],configuration file history>> we have now a native Git support to manage the configuration history. This feature goes a little farther thant the initial configuration file history in that it also manages content and all the configuration files (such as properties). This feature only work for **standalone servers** using the default directory layout.
+
+As mentioned in <<Command_line_parameters,Command line parameters>> we support the usage of a remote Git repository to pull the configuration from or create or use a local Git repository.
+In fact if a `.git` directory exists under `jboss.server.base.dir` then using Git for managing configuration files will be automatically activated.
+Each modification of the content or the configuration will result in a new commit when the operation is successful and there are changes to commit. If there is an authenticated user then it will be stored as the author of the commit.
+Please note that this is a real Git repository so using a native Git client you can manipulate it.
+
+Now if we execute a management operation that modifies
+the model, for example adding a new system property using the CLI:
+
+[source, ruby]
+----
+[standalone@localhost:9990 /] /system-property=test:add(value="test123")
+{"outcome" => "success"}
+----
+
+What happens is:
+
+* The change is applied to the configuration file.
+* The configuration file is added to a new commit.
+
+
+[[local_git_repository]]
+== Local Git Repository
+
+Starting the server with the option `--git-repository=local` will initiate a Git repository if none exists or use the current Git repository.When initiating the local Git repository a `.gitignore` file will be created and added to the initial commit.
+If a `--git-branch` parameter is added then the repository will create this branch and commit to it, otherwise it will default to `master`.
```

Comment:
I thought you were not creating branches in current implementation, at least it does not seem to work for me: --git-repo=local --git-branch=my_branch - the local repo is still on master, no branch is created - am I doing it wrong?

Reply:
no you are correct